src/main/java/com/ncc/aif/ValidateAIFCli.java

```diff
@@ -0,0 +1,597 @@
+    static final String ERR_SMALLER_THAN_MIN = "%s must be at least %d";
+    // Logging strings
+    static final String START_MSG = "AIF Validator";
+
+    //=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+    // Internal Constants
+    // ----------------------------
+    // Logger
+    private static final Logger logger = (Logger) (org.slf4j.LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME));
+    // ViolationThreshold
+    private static final String ABORT_PARAMETER_STRING = "Abort parameter";
+    private static final int DEFAULT_MAX_VIOLATIONS = -1;
+    private static final int MINIMUM_MAX_VIOLATIONS = 3;
+    // Profiling
+    private static final int LONG_QUERY_THRESH = 2000;
+    // Threading
+    private static final String THREAD_COUNT_STRING = "Thread count";
+    private static final int MINIMUM_THREAD_COUNT = 1;
+
+    //=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+    // Command Line Arguments
+    // ----------------------------
+    private int argCounter = 1;
+    //TODO: make ArgGroup when 4.0 is stable
+    @Option(names = "--ldc", description = "Validate against the LDC ontology")
+    private boolean useLDCOntology;
+
+    @Option(names = "--program", description = "Validate against the program ontology")
+    private boolean useProgramOntology;
+
+    @Option(names = "--ont", description = "Validate against the OWL-formatted ontolog(ies) at the specified filename(s)",
+            paramLabel = "FILE", arity = "1..*")
+    private List<File> customOntologies;
+
+    //TODO: make ArgGroup when 4.0 is stable
+    @Option(names = "--nist", description = "Validate against the NIST restrictions")
+    private boolean useNISTRestriction;
+
+    @Option(names = "--nist-ta3", description = "Validate against the NIST hypothesis restrictions (implies --nist)")
+    private boolean useNISTTA3Rescriction;
+
+    @Option(names = "-o", description = "Save validation report model to a file.  KB.ttl would result in KB-report.txt.")
+    private boolean outputToFile;
+
+    @Option(names = "--abort", description = "Abort validation after [num] SHACL violations (num > 2), or 3 violations if [num] is omitted.", paramLabel = "num")
+    private int maxValidationErrors = DEFAULT_MAX_VIOLATIONS;
+
+    @Option(names = "--pm", description = "Enable progress monitor that shows ongoing validation progress")
+    private boolean useProgressMonitor;
+
+    @Option(names = "-p", description = "Enable profiling", hidden = true)
+    private boolean useProfiling;
+
+    @Option(names = "--p2", description = "Enable progressive profiling", hidden = true)
+    private boolean useProgressiveProfiling;
+
+    @Option(names = "-t", description = "Specify the number of threads to use during validation", paramLabel = "num")
+    private int threads = MINIMUM_THREAD_COUNT;
+
+    //TODO: make ArgGroup when 4.0 is stable
+    @Option(names = "-d", description = "Validate all .ttl files in the specified directory", paramLabel = "DIRNAME")
+    private File directory;
```

Comment:
When I tried `validateAIF -f --abort --ldc`, I got, `Expected parameter for option '-f' but found '--abort'` and the usage info, which was reasonable behavior.

But when I tried `validateAIF -d --abort --ldc`, it started to try to validate, but failed with, `No files with .ttl suffix were specified.`.

I'm not sure if this is something you can fix-- and maybe switching to ArgGroups will fix this when 4.0 is stabilized.

Reply:
I'm not sure this will be fixed in 4.0, but the behavior with -d is a bug with picocli. We can get around this bug by making the -d switch act the same as -f. This would allow multiple directories, though.

Given a conversation with @nextcen-dgemoets, we're just going to leave it as-is for now